content/authoring/guidelines/submission-tests.md

```diff
@@ -42,7 +42,8 @@ Random tests are uncommon in "real life" coding and are somewhat specific to Cod
 - **Keep the amount of random tests reasonable if random tests are not used to verify the performance of the user's solution.** Keep their amount as small as possible as long as a good coverage is still guaranteed: in some situations, only a bunch of inputs are actually testable, so no need to tests each of them 10 times - use randomized testing instead. Otherwise, 100 random tests are generally enough, or maybe less, depending on the task/situation.
-- Under some rare circumstances, it is allowed to use so-called [**randomized tests**][randomized-tests] instead of fully random ones. For some types of problems (for example, kata simulating chess, grid puzzles like sudoku or nonograms) it may be difficult, or even impossible, to generate valid input configurations randomly. A problem with a small set of possible inputs does not need randomly generated inputs, if its whole input domain can be iterated. In such rare cases, it may be acceptable to use a predefined, hard-coded set of inputs (possibly with expected outputs too). Before the tests are run, the set of inputs should be shuffled or randomly sampled at each run to make the hard-coding of results more tedious for the user. If possible, some additional transformations can be randomly applied to the inputs if it can be easily accounted for in the result of the test (for example arrays can be reversed, game boards can be rotated, sides can be flipped, etc.)
+- For some types of problems (for example, kata simulating chess, or grid puzzles like sudoku or nonograms) it may be difficult, or even impossible, to generate valid input configurations randomly. In such rare cases, it may be acceptable to use a predefined, hard-coded set of inputs (possibly with expected outputs too). Before the tests are run, the set of inputs should be shuffled or randomly sampled at each run to make the hard-coding of results more tedious for the user. If possible, some additional transformations can be randomly applied to the inputs if it can be easily accounted for in the result of the test (for example arrays can be reversed, game boards can be rotated, sides can be flipped, etc.)
+- Some problems have a small set of possible inputs, and it's possible to easily enumerate their whole input domain. For this kind of tasks, it's not necessary to randomly generate inputs. It is allowed to pregenerate or hard-code all possible inputs, and before the tests are run, shuffle or randomly sample them at each run to render solutions which abuse order of inputs infeasible.
```

Comment:
- `For this kind of tasks` -> `For these kinds of tasks`
- I'd write `pregenerate` as `pre-generate`

Reply:
```suggestion
- Some problems have a small set of possible inputs, and it's possible to easily enumerate their whole input domain. For these kinds of tasks, it's not necessary to randomly generate inputs. It is allowed to pre-generate or hard-code all possible inputs, and before the tests are run, shuffle or randomly sample them at each run to render solutions which abuse order of inputs infeasible.
```